### ISSUE TITLE
Container startup fails when using bind mounts on /opt/oracle/admin

### ISSUE THREAD

**Issue by rgibbard** [gvenzl/oci-oracle-xe#198]
```
services:
  database:
    image: gvenzl/oracle-xe:latest
    container_name: oracledb
    environment:
      - ORACLE_PASSWORD=manager
    ports:
      - 1521:1521
      - 8080:8080
      - 5500:5500
    volumes:
      - ./oradata:/opt/oracle/oradata                     # persistent oracle database data
      - ./oradata/admin:/opt/oracle/admin            # persistent oracle admin directory (datapump dir, etc..)
```

```
$ ls
docker-compose.yml      oradata                 scripts
$ docker-compose up
[+] Running 2/2
 ⠿ Network test_default  Created                                                        0.0s
 ⠿ Container oracledb    Created                                                        0.1s
Attaching to oracledb
oracledb  | CONTAINER: starting up...
oracledb  | CONTAINER: first database startup, initializing...
oracledb  | CONTAINER: uncompressing database data files, please wait...
oracledb  | CONTAINER: done uncompressing database data files, duration: 63 seconds.
oracledb  | CONTAINER: starting up Oracle Database...
oracledb  | 
oracledb  | LSNRCTL for Linux: Version 21.0.0.0.0 - Production on 28-MAR-2023 18:20:59
oracledb  | 
oracledb  | Copyright (c) 1991, 2021, Oracle.  All rights reserved.
oracledb  | 
oracledb  | Starting /opt/oracle/product/21c/dbhomeXE/bin/tnslsnr: please wait...
oracledb  | 
oracledb  | TNSLSNR for Linux: Version 21.0.0.0.0 - Production
oracledb  | System parameter file is /opt/oracle/homes/OraDBHome21cXE/network/admin/listener.ora
oracledb  | Log messages written to /opt/oracle/diag/tnslsnr/6ea17add153e/listener/alert/log.xml
oracledb  | Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
oracledb  | Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
oracledb  | 
oracledb  | Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC_FOR_XE)))
oracledb  | STATUS of the LISTENER
oracledb  | ------------------------
oracledb  | Alias                     LISTENER
oracledb  | Version                   TNSLSNR for Linux: Version 21.0.0.0.0 - Production
oracledb  | Start Date                28-MAR-2023 18:20:59
oracledb  | Uptime                    0 days 0 hr. 0 min. 0 sec
oracledb  | Trace Level               off
oracledb  | Security                  ON: Local OS Authentication
oracledb  | SNMP                      OFF
oracledb  | Default Service           XE
oracledb  | Listener Parameter File   /opt/oracle/homes/OraDBHome21cXE/network/admin/listener.ora
oracledb  | Listener Log File         /opt/oracle/diag/tnslsnr/6ea17add153e/listener/alert/log.xml
oracledb  | Listening Endpoints Summary...
oracledb  |   (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
oracledb  |   (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
oracledb  | The listener supports no services
oracledb  | The command completed successfully
oracledb  | ORA-09925: Unable to create audit trail file
oracledb  | Linux-x86_64 Error: 2: No such file or directory
oracledb  | Additional information: 9925
oracledb exited with code 197
```

This docker-compose configuration works fine when using the official Oracle docker images.

**Comment by gvenzl** [comment]
Hi @rgibbard,

Thanks a lot for using these images!

From experience, this is most likely a permission issue on the `admin` folder, I have seen `ORA-09925: Unable to create audit trail file` before when the folder was either missing or permissions set that didn't allow the internal `oracle` user to write to the directory.
However, permission issues would be the same for `/opt/oracle/oradata`, does the container come up if you just map `- ./oradata:/opt/oracle/oradata` and not `- ./oradata/admin:/opt/oracle/admin`?

**Comment by rgibbard** [comment]
@gvenzl Yes that is correct. If only `/opt/oracle/oradata` is mapped, its fine. If you add in `/opt/oracle/admin` it fails.  This configuration works fine with the official Oracle Database images. 

This is the directory structure I am using prior to `docker-compose up`:
```
$ ls -lR
total 8
-rw-r--r--@ 1 user  staff  415 Apr 10 09:46 docker-compose.yml
drwxr-xr-x@ 3 user  staff   96 Apr 10 09:47 oradata

./oradata:
total 0
drwxr-xr-x@ 2 user  staff  64 Apr 10 09:47 admin

./oradata/admin:
total 0
```

##### `docker-compose.yml`
```
services:
  database:
    image: gvenzl/oracle-xe:latest
    container_name: oracledb
    environment:
      - ORACLE_PASSWORD=manager
    ports:
      - 1521:1521
      - 8080:8080
      - 5500:5500
    volumes:
      - ./oradata:/opt/oracle/oradata               # persistent oracle database data
      - ./oradata/admin:/opt/oracle/admin      # persistent oracle admin directory (datapump dir, etc..)
```

**Comment by gvenzl** [comment]
Hey @rgibbard,

I see now what's going on there, thanks a lot for the additional info.
I think you are running into the ever-annoying Docker issue that what they used to call host-only volumes (when you point them to a specific location on the file system) do not copy anything from inside the container into that volume, but literally just mount over them.

The above output shows that you have an empty admin folder:

```
$ ls -lR
total 8
-rw-r--r--@ 1 user  staff  415 Apr 10 09:46 docker-compose.yml
drwxr-xr-x@ 3 user  staff   96 Apr 10 09:47 oradata

./oradata:
total 0
drwxr-xr-x@ 2 user  staff  64 Apr 10 09:47 admin

./oradata/admin:
total 0
```

And I bet you when you go into the container, that admin folder will also be empty, e.g. (on Podman here):

```
[gvenzl@localhost test]$ mkdir admin
[gvenzl@localhost test]$ ls -al admin
total 0
drwxr-xr-x. 2 gvenzl gvenzl  6 May 13 11:39 .
drwxr-xr-x. 3 gvenzl gvenzl 45 May 13 11:39 ..
[gvenzl@localhost test]$ podman run --name test -v ./admin:/opt/oracle/admin --entrypoint=bash -ti -u=0 --privileged gvenzl/oracle-xe
bash-4.4# ls -al
total 429344
drwxr-xr-x.  1 oracle oinstall        20 Mar 12 22:54 .
drwxr-xr-x.  1 root   root            20 Mar 12 22:52 ..
-rw-r--r--.  1 oracle oinstall       310 Mar 12 22:43 .bash_profile
-rw-r--r--.  1 oracle oinstall 439471449 Mar 12 22:51 XE.zip
drwxr-xr-x.  2 root   root             6 May 13 18:39 admin
drwxr-x---.  3 oracle oinstall        16 Mar 12 22:37 audit
drwxr-x---.  5 oracle oinstall        46 Mar 12 22:36 cfgtoollogs
-rwxrwx---.  1 oracle oinstall     16530 Feb  6 06:20 container-entrypoint.sh
-rwxrwx---.  1 oracle oinstall      1801 Apr 23  2022 createAppUser
drwxr-x---.  2 oracle oinstall        88 Mar 12 22:50 dbs
drwxrwxr-x. 23 oracle oinstall      4096 Mar 12 22:36 diag
-rwxrwx---.  1 oracle oinstall      1042 Nov 15  2021 healthcheck.sh
drwxr-x---.  3 oracle oinstall        28 Mar 12 22:36 homes
-rwxrwx---.  1 oracle oinstall    131116 Mar 12 01:38 install.2130.sh
drwxrwx---.  4 oracle oinstall        78 Mar 12 22:43 oraInventory
drwxr-x---.  3 oracle oinstall        16 Mar 12 22:51 oradata
drwxr-xr-x.  1 oracle oinstall        17 Mar 12 22:33 product
-rwxrwx---.  1 oracle oinstall      1066 Jul 11  2021 resetPassword
bash-4.4# ls -al admin
total 0
drwxr-xr-x. 2 root   root      6 May 13 18:39 .
drwxr-xr-x. 1 oracle oinstall 20 Mar 12 22:54 ..
```

However, the `admin` folder has several subfolders in it, and that's where Oracle is trying to write to:

```
[gvenzl@localhost test]$ podman run --name test --entrypoint=bash -ti gvenzl/oracle-xe
bash-4.4$ ls -al
total 429344
drwxr-xr-x.  1 oracle oinstall        20 Mar 12 22:54 .
drwxr-xr-x.  1 root   root            20 Mar 12 22:52 ..
-rw-r--r--.  1 oracle oinstall       310 Mar 12 22:43 .bash_profile
-rw-r--r--.  1 oracle oinstall 439471449 Mar 12 22:51 XE.zip
drwxr-x---.  3 oracle oinstall        16 Mar 12 22:36 admin
drwxr-x---.  3 oracle oinstall        16 Mar 12 22:37 audit
drwxr-x---.  5 oracle oinstall        46 Mar 12 22:36 cfgtoollogs
-rwxrwx---.  1 oracle oinstall     16530 Feb  6 06:20 container-entrypoint.sh
-rwxrwx---.  1 oracle oinstall      1801 Apr 23  2022 createAppUser
drwxr-x---.  2 oracle oinstall        88 Mar 12 22:50 dbs
drwxrwxr-x. 23 oracle oinstall      4096 Mar 12 22:36 diag
-rwxrwx---.  1 oracle oinstall      1042 Nov 15  2021 healthcheck.sh
drwxr-x---.  3 oracle oinstall        28 Mar 12 22:36 homes
-rwxrwx---.  1 oracle oinstall    131116 Mar 12 01:38 install.2130.sh
drwxrwx---.  4 oracle oinstall        78 Mar 12 22:43 oraInventory
drwxr-x---.  3 oracle oinstall        16 Mar 12 22:51 oradata
drwxr-xr-x.  1 oracle oinstall        17 Mar 12 22:33 product
-rwxrwx---.  1 oracle oinstall      1066 Jul 11  2021 resetPassword
bash-4.4$ ls -al admin
total 0
drwxr-x---. 3 oracle oinstall 16 Mar 12 22:36 .
drwxr-xr-x. 1 oracle oinstall 20 Mar 12 22:54 ..
drwxr-x---. 6 oracle oinstall 64 Mar 12 22:36 XE
bash-4.4$ ls -lR admin/
admin/:
total 0
drwxr-x---. 6 oracle oinstall 64 Mar 12 22:36 XE

admin/XE:
total 0
drwxr-x---. 2 oracle oinstall  6 Mar 12 22:51 adump
drwxr-x---. 2 oracle oinstall  6 Mar 12 22:51 dpdump
drwxr-x---. 2 oracle oinstall 36 Mar 12 22:40 pfile
drwxr-x---. 2 oracle oinstall 44 Mar 12 22:36 xdb_wallet

admin/XE/adump:
total 0

admin/XE/dpdump:
total 0

admin/XE/pfile:
total 4
-rw-r-----. 1 oracle oinstall 2052 Mar 12 22:36 init.ora.2122023224053

admin/XE/xdb_wallet:
total 8
-rw-------. 1 oracle oinstall 3864 Mar 12 22:36 cwallet.sso
-rw-------. 1 oracle oinstall 3819 Mar 12 22:36 ewallet.p12
```

Can you change your docker-compose file to use named volumes instead and try again:

```
volumes:
  oraadmin:
  oradata:
services:
  database:
    image: gvenzl/oracle-xe:latest
    container_name: oracledb
    environment:
      - ORACLE_PASSWORD=manager
    ports:
      - 1521:1521
      - 8080:8080
      - 5500:5500
    volumes:
      - oradata:/opt/oracle/oradata               # persistent oracle database data
      - oraadmin:/opt/oracle/admin
```

In my case, that works as expected:

```
[gvenzl@localhost test]$ podman-compose up
[database] | CONTAINER: starting up...
[database] | CONTAINER: first database startup, initializing...
[database] | CONTAINER: uncompressing database data files, please wait...
[database] | CONTAINER: done uncompressing database data files, duration: 25 seconds.
[database] | CONTAINER: starting up Oracle Database...
[database] |
[database] | LSNRCTL for Linux: Version 21.0.0.0.0 - Production on 13-MAY-2023 18:43:01
[database] |
[database] | Copyright (c) 1991, 2021, Oracle.  All rights reserved.
[database] |
[database] | Starting /opt/oracle/product/21c/dbhomeXE/bin/tnslsnr: please wait...
[database] |
[database] | TNSLSNR for Linux: Version 21.0.0.0.0 - Production
[database] | System parameter file is /opt/oracle/homes/OraDBHome21cXE/network/admin/listener.ora
[database] | Log messages written to /opt/oracle/diag/tnslsnr/0d0c5897bc85/listener/alert/log.xml
[database] | Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
[database] | Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
[database] |
[database] | Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC_FOR_XE)))
[database] | STATUS of the LISTENER
[database] | ------------------------
[database] | Alias                     LISTENER
[database] | Version                   TNSLSNR for Linux: Version 21.0.0.0.0 - Production
[database] | Start Date                13-MAY-2023 18:43:01
[database] | Uptime                    0 days 0 hr. 0 min. 0 sec
[database] | Trace Level               off
[database] | Security                  ON: Local OS Authentication
[database] | SNMP                      OFF
[database] | Default Service           XE
[database] | Listener Parameter File   /opt/oracle/homes/OraDBHome21cXE/network/admin/listener.ora
[database] | Listener Log File         /opt/oracle/diag/tnslsnr/0d0c5897bc85/listener/alert/log.xml
[database] | Listening Endpoints Summary...
[database] |   (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
[database] |   (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
[database] | The listener supports no services
[database] | The command completed successfully
[database] | ORACLE instance started.
[database] |
[database] | Total System Global Area 1241512272 bytes
[database] | Fixed Size		    9685328 bytes
[database] | Variable Size		  603979776 bytes
[database] | Database Buffers	  620756992 bytes
[database] | Redo Buffers		    7090176 bytes
[database] | Database mounted.
[database] | Database opened.
[database] |
[database] | CONTAINER: Resetting SYS and SYSTEM passwords.
[database] |
[database] | User altered.
[database] |
[database] |
[database] | User altered.
[database] |
[database] |
[database] | #########################
[database] | DATABASE IS READY TO USE!
[database] | #########################
[database] |
[database] | ##################################################################
[database] | CONTAINER: The following output is now from the alert_XE.log file:
[database] | ##################################################################
[database] | 2023-05-13T18:43:13.995084+00:00
[database] | XEPDB1(3):SUPLOG: Set PDB SUPLOG SGA at PDB OPEN, old 0x18, new 0x0 (no suplog)
[database] | 2023-05-13T18:43:14.883445+00:00
[database] | XEPDB1(3):Opening pdb with Resource Manager plan: DEFAULT_PLAN
[database] | 2023-05-13T18:43:15.042934+00:00
[database] | Pluggable database XEPDB1 opened read write
[database] | Starting background process CJQ0
[database] | 2023-05-13T18:43:15.155212+00:00
[database] | CJQ0 started with pid=56, OS id=202
[database] | Completed: ALTER DATABASE OPEN
```
